### PR TITLE
Sept 2023 updates

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**

https://github.com/fivetran/dbt_klaviyo/issues/28

**This PR will result in the following new package version:**

v0.7.0

Removing deprecated and adding new fields

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

Following these [release notes](https://fivetran.com/docs/applications/klaviyo/changelog) for changes made to the connector.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [ ] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [ ] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [ ] You have provided details below around the validation steps performed to gain confidence in these changes.

- dbt docs show updated ymls
- downstream models succeed 

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [ ] README updates have been applied (if applicable)
    
- [ ] DECISIONLOG updates have been updated (if applicable)
- [ ] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [ ] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

**Comment by fivetran-reneeli** [comment]
Will wait on regen docs til after PR approved

File: models/stg_klaviyo__campaign.sql [hunk 1]
```diff
@@ -31,14 +31,15 @@ final as (
...
         cast(id as {{ dbt.type_string() }} ) as campaign_id,
-        is_segmented,
         name as campaign_name,
         send_time as scheduled_to_send_at,
         sent_at,
         coalesce(status, lower(status_label)) as status,
         status_id,
         subject,
         updated as updated_at,
+        archived,
```
Comment (fivetran-jamie) [inline models/stg_klaviyo__campaign.sql:41]: since this is a boolean do we wanna rename it to `is_archived`?

File: models/stg_klaviyo__flow.sql [hunk 2]
```diff
@@ -29,14 +29,9 @@ final as (
...
         status,
-        {% if target.type == 'snowflake'%}
-        "TRIGGER" 
-        {% else %}
-        trigger
-        {% endif %}
-        as flow_trigger,
         updated as updated_at,
-        customer_filter as person_filter,
+        archived,
```
Comment (fivetran-jamie) [inline models/stg_klaviyo__flow.sql:33]: same comment as above -- should we rename to `is_archived`?

File: models/stg_klaviyo__person.sql [hunk 3]
```diff
@@ -42,6 +42,7 @@ final as (
         timezone,
         title,
         updated as updated_at,
+        last_event_date as last_event_date_at,
```
Comment (fivetran-jamie) [inline models/stg_klaviyo__person.sql:45]: is the `date_at` part redundant? wondering if we should leave it as `last_event_date` or rename it to just `last_event_at`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+
+- We have added these columns from the following tables:
+
+| **Table**                          | **Column**                                                                                                                                                                                                                             |
+|--------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| CAMPAIGN | archived
+| CAMPAIGN | scheduled
+| FLOW | archived
+| FLOW | trigger_type
+| PERSON | last_event_date
```
Comment (fivetran-jamie) [inline CHANGELOG.md:22]: i feel like we should have a third column that shows what we rename these fields to (if we rename them) in the respective staging models

Comment (fivetran-jamie) [reply CHANGELOG.md:22]: also could be good to add DAG links to the affected source tables so people can quickly see the descriptions for these new fields

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,29 @@
+# dbt_klaviyo_source v0.7.0
+[PR #18](https://github.com/fivetran/dbt_klaviyo_source/pull/18/) includes updates regarding the [September 2023](https://fivetran.com/docs/applications/klaviyo/changelog#september2023) changes to the Klaviyo connector.
+
+## 🚨 Breaking Changes 🚨:
+
+- We have removed these columns from the following tables:
```
Comment (fivetran-jamie) [inline CHANGELOG.md:6]:
```suggestion
- We have removed these deprecated columns from the following tables:
```

**Review by fivetran-jamie — CHANGES_REQUESTED** [review]
looks pretty good! nothing big stands out, but left some style- and doc-related comments

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-jamie) [reply CHANGELOG.md:22]: same for the removed fields

File: models/stg_klaviyo__campaign.sql (hunk 1 above)
Comment (fivetran-reneeli) [reply models/stg_klaviyo__campaign.sql]: Ah thanks Jamie-- just realized I had some changes that weren't pushed, this included

File: models/stg_klaviyo__flow.sql (hunk 2 above)
Comment (fivetran-reneeli) [reply models/stg_klaviyo__flow.sql]: udpated

File: models/stg_klaviyo__person.sql (hunk 3 above)
Comment (fivetran-reneeli) [reply models/stg_klaviyo__person.sql:45]: good idea, renamed to last_event_at

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md]: awesome, updated!

**Review by fivetran-jamie — COMMENTED** [review]
looks great! will approve after some minor things in the changelog are addressed and docs are regenerated

File: models/stg_klaviyo__person.sql [hunk 6]
```diff
@@ -42,6 +42,7 @@ final as (
         timezone,
         title,
         updated as updated_at,
+        last_event_date as last_event_at,
```
Comment (fivetran-jamie) [inline models/stg_klaviyo__person.sql:45]: ah actually this is causing ambiguous column errors in the final persons model due to https://github.com/fivetran/dbt_klaviyo/blob/main/models/intermediate/int_klaviyo__person_metrics.sql#L17

may want to leave it as `last_event_date` or consider removing it?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,35 @@
...
+
+- We have removed these deprecated columns from the following tables:
+
+| **Table**                          | **Column**                                                                                                                                                                                                                             |
+|--------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| CAMPAIGN | is_segmented
+| FLOW | customer_filter
+| FLOW | trigger
+
+- We have added these columns from the following tables:
```
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply CHANGELOG.md:14]: ah good reminder

File: models/stg_klaviyo__person.sql (hunk 6 above)
Comment (fivetran-reneeli) [reply models/stg_klaviyo__person.sql:45]: Oh good noticing, I didn't realize the downstream error. I feel like it's still worthwhile adding it, so will just keep it as `last_event_date`. Thanks Jamie!

**Comment by fivetran-reneeli** [comment]
Thanks @fivetran-jamie! Updated everything and regen docs. I also realized that email in the persons staging may not necessarily exist, based off testing off the internal schema, so I removed the not null test.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,36 @@
...
+
+- We have added these columns from the following tables and renamed them:
+
+| **Table**       | **Column**  | **New Name**       |
+| :---        |    :----:   |          ---: |
+| CAMPAIGN | archived | is_archived
+| CAMPAIGN | scheduled | scheduled_at
+| FLOW | archived | is_archived
+| FLOW | trigger_type | trigger_type
+| PERSON | last_event_date | last_event_at
```
Comment (fivetran-jamie) [inline CHANGELOG.md:22]:
```suggestion
| PERSON | last_event_date | last_event_date
```

**Review by fivetran-jamie — APPROVED** [review]
lgtm! just one thing in the changelog but otherwise this is approved

**Pull request merged**